cmake/schemes/url_sha1_odb-sqlite_autotools.cmake.in

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2013, 2015 Ruslan Baratov, Alexandre Pretyman
+# All rights reserved.
+
+cmake_minimum_required(VERSION 3.0)
+project(Hunter)
+
+include(ExternalProject) # ExternalProject_Add
+
+list(APPEND CMAKE_MODULE_PATH "@HUNTER_SELF@/cmake/modules")
+
+include(hunter_fatal_error)
+include(hunter_status_debug)
+include(hunter_test_string_not_empty)
+
+include(hunter_autotools_append_toolchain_binaries)
+include(hunter_autotools_append_preprocessor_flags)
+include(hunter_autotools_append_compiler_cxxflags)
+include(hunter_autotools_append_compiler_cflags)
+include(hunter_autotools_append_linker_flags)
+
+hunter_status_debug("Scheme: url_sha1_odb-sqlite_autotools")
+
+set(HUNTER_STATUS_DEBUG "@HUNTER_STATUS_DEBUG@")
+
+if("@MSVC@")
+  hunter_fatal_error("Autotools scheme not supported with Visual Studio")   
+endif()
+
+list(APPEND CMAKE_MODULE_PATH "@HUNTER_SELF@/cmake/find")
+
+set(ODB_ROOT "@ODB_ROOT@")
+find_package(odb REQUIRED)
+
+set(SQLITE3_ROOT "@SQLITE3_ROOT@")
+find_package(SQLite3 REQUIRED)
+
```

Comment:
I think I got Chip and Dale (my two neurons) busy working on this. 

The API that would be best would be a *function* like:
```cmake
hunter_autotools_configure_options(configure_opts
    CPPFLAGS "-DEXAMPLE -I/usr/local/this/and/that"
    CXXFLAGS "-extra-stuff"
    CFLAGS "-other-flags"
    LDFLAGS "-L/usr/local/lib -llibrary"
    EXTRAFLAGS "--enable-feature"
)
```

`hunter_autotools_configure_options(...)` would add the toolchain binaries and cross compiling host to `configure_opts`
 
@ruslo  what I need help deciding is:
1. let be the responsibility of the caller to add `${CMAKE_CXX_FLAGS}`.
2. `${CMAKE_CXX_FLAGS}` gets added automatically from inside the function.

`LDFLAGS` is more tricky because depending what the autotools project generates, it could have different linker flags, or so I believe, but I could be very wrong, since I haven't generated a shared object in years, not sure how different it is from static library.